### PR TITLE
accept query parameters (for tags and categories)

### PR DESCRIPTION
**Pull request by AmarOk1412**
Cf. issue #25 

Add 3 parameters. Now we can get http://firefox-dev.tools/?easy=true&mentored=true&category=inspector for example.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -147,6 +170,10 @@ function createToolListMarkup(parentEl) {
       }
     });
 
+    if(categorieInUrl === keys[i]) {
```
Comment (nt1m) [inline app.js]: nits: space after if

File: app.js [hunk 2]
```diff
@@ -134,6 +156,7 @@ function getToolTooltip(id) {
 
 function createToolListMarkup(parentEl) {
   var keys = Object.keys(COMPONENT_MAPPING);
+  var categorieInUrl = getParameterByName("category");
```
Comment (nt1m) [inline app.js]: nit: categoryInUrl

File: app.js [hunk 3]
```diff
@@ -13,6 +13,28 @@ function hasFilter(name, filters) {
   return false;
 }
 
+function getParameterByName(name, url) {
+  if (!url) url = window.location.href;
```
Comment (nt1m) [inline app.js]: I would prefer getting the parameters from window.location.search rather than href.

File: app.js [hunk 4]
```diff
@@ -13,6 +13,28 @@ function hasFilter(name, filters) {
...
+  name = name.replace(/[\[\]]/g, "\\$&");
+  var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
+  results = regex.exec(url);
+  if (!results) return null;
+  if (!results[2]) return '';
+  return decodeURIComponent(results[2].replace(/\+/g, " "));
+}
+
+function parseTagsInUrl(url) {
+  if (!url) url = window.location.search;
```
Comment (nt1m) [inline app.js]: Can you remove this line and the url parameter in this function?

File: app.js [hunk 5]
```diff
@@ -13,6 +13,28 @@ function hasFilter(name, filters) {
   return false;
 }
 
+function getParameterByName(name, url) {
```
Comment (nt1m) [inline app.js]: The url parameter is unused in the file, can you remove it?

File: app.js [hunk 6]
```diff
@@ -13,6 +13,28 @@ function hasFilter(name, filters) {
...
+  if (!url) url = window.location.search;
```
Comment (nt1m) [inline app.js]: var urlParams = window.location.search;

File: app.js [hunk 7]
```diff
@@ -13,6 +13,28 @@ function hasFilter(name, filters) {
...
+  name = name.replace(/[\[\]]/g, "\\$&");
```
Comment (nt1m) [inline app.js]: nit: add a new line after this line

File: app.js [hunk 8]
```diff
@@ -13,6 +13,28 @@ function hasFilter(name, filters) {
...
+  var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
+  results = regex.exec(url);
```
Comment (nt1m) [inline app.js]: nit: use `var results` (we don't use the var x, y; notation)
and add a new line after this one.

**Review by nt1m — CHANGES_REQUESTED** [review]
Looks good so far! Thanks for the PR

File: app.js (hunk 4 above)
Comment (AmarOk1412) [reply app.js]: Woops, my bad

**Comment by nt1m** [comment]
I'll let @captainbrosset do a last review.

File: app.js [hunk 9]
```diff
@@ -13,6 +13,28 @@ function hasFilter(name, filters) {
   return false;
 }
 
+function getParameterByName(name) {
+  var urlParams = window.location.search;
+  name = name.replace(/[\[\]]/g, "\\$&");
+
+  var results = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)").exec(urlParams);
```
Comment (captainbrosset) [inline app.js]: No need for parsing url parameters, the `URL` constructor can be used instead:

```
var params = new URL(window.location).searchParams;
```

This gives you a ready to use object to retrieve the parameters. See https://developer.mozilla.org/en-US/docs/Web/API/URL

It seems to have decent browser support too except for Edge/IE, but I've never tested the site in these browsers.

File: app.js [hunk 10]
```diff
@@ -134,6 +156,7 @@ function getToolTooltip(id) {
 
 function createToolListMarkup(parentEl) {
   var keys = Object.keys(COMPONENT_MAPPING);
+  var categoryInUrl = getParameterByName("category");
```
Comment (captainbrosset) [inline app.js]: I would rename `category` to `tool`. We don't use the word category anywhere in the code or the UI, and tool seems to me like a better fit. Also easier to type.

File: app.js [hunk 11]
```diff
@@ -13,6 +13,28 @@ function hasFilter(name, filters) {
...
+  name = name.replace(/[\[\]]/g, "\\$&");
+
+  var results = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)").exec(urlParams);
+  
+  if (!results) return null;
+  if (!results[2]) return '';
+  return decodeURIComponent(results[2].replace(/\+/g, " "));
+}
+
+function parseTagsInUrl() {
```
Comment (captainbrosset) [inline app.js]: This function doesn't parse tags. It retrieves them form the URL using another function and then sets the corresponding inputs.
So I suggest changing its name to `setFiltersFromUrlParams`.

File: app.js [hunk 12]
```diff
@@ -13,6 +13,28 @@ function hasFilter(name, filters) {
...
+  var easy = getParameterByName("easy");
+  var mentored = getParameterByName("mentored");
+  if (easy === "false" || (mentored === "true" && easy === null)) {
```
Comment (captainbrosset) [inline app.js]: About this logic: why do we need to even have values for `easy` and `mentored`.
It seems to me that this could work with just `?easy&mentored`, or `?easy`, or `?mentored`.
I don't think we need them to have `true/false` as a value. It makes the URL harder to type, and we can derive the same meaning with the values.

So:
- if neither `easy` nor `mentored` are part of the URL params, then we use to the default settings,
- if `easy` is in the URL params but not `mentored`, then we search for easy/non-mentored bugs 
- if `easy` is not in the URL params but `mentored` is, then we search for non-easy/mentored bugs
- if both `easy` and `mentored` are in the URL params, then we search for easy/mentored bugs

How does that sound?

**Review by captainbrosset — CHANGES_REQUESTED** [review]
Thanks for the PR.
Here are a few more comments.

File: app.js (hunk 9 above)
Comment (AmarOk1412) [reply app.js]: Awesome! It's a lot better than a Regex

File: app.js (hunk 12 above)
Comment (AmarOk1412) [reply app.js]: I agree, it's easier to write.

File: app.js [hunk 13]
```diff
@@ -14,24 +14,36 @@ function hasFilter(name, filters) {
...
-  return decodeURIComponent(results[2].replace(/\+/g, " "));
+  var params = new URL(window.location).searchParams;
+  var value = params.get(name);
+
+  if (value) {
+    return decodeURIComponent(value.replace(/\+/g, " "));
+  }
+
+  if (params.has(name)) {
+    return '';
```
Comment (nt1m) [inline app.js]: Nit: use double quotes

File: app.js [hunk 14]
```diff
@@ -303,7 +342,7 @@ function matchesSearchString(bug) {
   var query = searchString.toLowerCase();
 
   return bug.summary.toLowerCase().indexOf(query) !== -1 ||
-         (bug.id + "").indexOf(query) !== -1;
+  (bug.id + "").indexOf(query) !== -1;
```
Comment (captainbrosset) [inline app.js]: nit: this seems like an unrelated formatting change. Can you change it back?

File: app.js [hunk 15]
```diff
@@ -280,7 +319,7 @@ function createBugMarkup(bug) {
       attributes: {
         "class": "tag old-bug",
         "title": "This bug has been inactive for more than " +
-                 INACTIVE_AFTER + " days"
+        INACTIVE_AFTER + " days"
```
Comment (captainbrosset) [inline app.js]: nit: this seems like an unrelated formatting change. Can you change it back?

File: app.js [hunk 16]
```diff
@@ -24,13 +58,13 @@ function getSearchParams(options) {
...
-                       "keywords",
-                       "mentors",
-                       "attachments"],
+    "assigned_to",
+    "summary",
+    "last_change_time",
+    "component",
+    "keywords",
+    "mentors",
+    "attachments"],
```
Comment (captainbrosset) [inline app.js]: nit: this seems like an unrelated formatting change. Can you change it back?

**Review by captainbrosset — APPROVED** [review]
Thanks for the new commits. Just a few final nit comments, I'm not blocking the review just for this, but if you do have time, it'd gladly take another commit with these fixed.

**Comment by captainbrosset** [comment]
Thanks @AmarOk1412
